taichi/rhi/cuda/cuda_driver.cpp

```diff
@@ -157,13 +181,60 @@ CUBLASDriver &CUBLASDriver::get_instance() {
   return *instance;
 }
 
+std::string get_lib_name_linux(const std::string &lib_name, int version) {
+  return "lib" + lib_name + ".so." + std::to_string(version);
+}
+
+std::string get_lib_name_windows(const std::string &lib_name,
+                                 const std::string &win_arch_name,
+                                 int version) {
+  return lib_name + win_arch_name + std::to_string(version) + ".dll";
+}
+
+bool CUDADriverBase::try_load_lib_any_version(
+    const std::string &lib_name,
+    const std::string &win_arch_name,
+    const std::vector<int> &versions_to_try) {
+  // Check if any versions of this lib are already loaded.
+  for (auto version : versions_to_try) {
+    std::string lib_name_linux = get_lib_name_linux(lib_name, version);
+    std::string lib_name_windows =
+        get_lib_name_windows(lib_name, win_arch_name, version);
+    if (check_lib_loaded(lib_name_linux, lib_name_windows)) {
+      load_lib(lib_name_linux, lib_name_windows);
+      return true;
+    }
+  }
+
+  // Try load any version of this lib if none of them are loaded.
+  bool loaded = false;
+  if (!loaded) {
+#ifdef WIN32
+    for (auto version : versions_to_test) {
+      std::string lib_name_windows = get_lib_name_windows(lib_name, version);
+      loader_ = std::make_unique<DynamicLoader>(lib_name_linux);
+      loaded = loader_->loaded();
+      if (loaded) {
+        break;
+      }
+    }
+#else
+    // Use the default version on linux.
+    std::string lib_name_linux = "lib" + lib_name + ".so";
+    loader_ = std::make_unique<DynamicLoader>(lib_name_linux);
+    loaded = loader_->loaded();
+#endif
+  }
+  return loaded;
+}
+
 bool CUBLASDriver::load_cublas() {
   /* To be compatible with torch environment, please libcublas.so.11 other than
    * libcublas.so. When using libcublas.so, the system cublas will be loaded and
    * it would confict with torch's cublas. When using libcublas.so.11, the
    * torch's cublas will be loaded.
    */
-  cublas_loaded_ = load_lib("libcublas.so.11", "cublas64_11.dll");
+  cublas_loaded_ = try_load_lib_any_version("cublas", "64_", {12, 11, 10});
```

Comment:
The function looks weird and overly entangled with current problem, it could have to change with the requirements if it exhibits even a tiny change, IMHO.

We should consult another reviewer.

Reply:
Here just trying to load the cublas of different versions (12/11/10), I believe the related comments are enough to let other contributors know what we are doing here.

Reply:
Could we use `{11, 12, 10}` since PyTorch uses 11 now?